pkg/registry/apis/provisioning/worker_export.go

```diff
@@ -0,0 +1,234 @@
+package provisioning
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"path"
+	"path/filepath"
+
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	apiutils "github.com/grafana/grafana/pkg/apimachinery/utils"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/auth"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/jobs"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/repository"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
+	"gopkg.in/yaml.v3"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/dynamic"
+)
+
+var _ jobs.Worker = (*exportWorker)(nil)
+
+type exportWorker struct {
+	getter         RepoGetter
+	resourceClient *resources.ClientFactory
+	identities     auth.BackgroundIdentityService
+	logger         *slog.Logger
+}
+
+func (w *exportWorker) Supports(ctx context.Context, job *provisioning.Job) bool {
+	return job.Spec.Action == provisioning.JobActionExport
+}
+
+func (w *exportWorker) Process(ctx context.Context, job provisioning.Job) (*provisioning.JobStatus, error) {
+	logger := w.logger.With("job", job.Name, "namespace", job.GetNamespace())
```

Comment:
What do you think about moving the logic into the `Replicator` as well? To make it similar to the new `Sync`.

Reply:
Done!